reference/docker/reference.go

```diff
@@ -338,7 +338,7 @@ func WithDigest(name Named, digest digest.Digest) (Canonical, error) {
 
 // TrimNamed removes any tag or digest from the named reference.
 func TrimNamed(ref Named) Named {
-	domain, path := SplitHostname(ref)
+	domain, path := Domain(ref), Path(ref)
```

Comment:
For functions in the same package, it probably just makes sense to use `splitDomain` rather than make two calls here